microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java

```diff
@@ -16,56 +16,430 @@
+        }
+
+        // this is not needed at runtime, collect garbage
+        serverConfigBuilder = null;
+        routingBuilder = null;
+        namedRoutings = null;
+    }
+
+    private void registerJaxRsApplications(BeanManager beanManager, ServerConfiguration serverConfig) {
+        JaxRsCdiExtension jaxRs = beanManager.getExtension(JaxRsCdiExtension.class);
+
+        jaxRs.applicationsToRun()
+                .forEach(it -> addApplication(serverConfig, jaxRs, it));
+    }
+
+    private void registerDefaultRedirect() {
+        Optional.ofNullable(basePath)
+                .or(() -> config.get("server.base-path").asString().asOptional())
+                .ifPresent(basePath -> {
+                    routingBuilder.any("/", (req, res) -> {
+                        res.status(Http.Status.MOVED_PERMANENTLY_301);
+                        res.headers().put(Http.Header.LOCATION, basePath);
+                        res.send();
+                    });
+                });
+    }
+
+    private void registerStaticContent() {
+        Config config = (Config) ConfigProvider.getConfig();
+        config = config.get("server.static");
+
+        config.get("classpath")
+                .ifExists(this::registerClasspathStaticContent);
+
+        config.get("path")
+                .ifExists(this::registerPathStaticContent);
+    }
+
+    private void registerPathStaticContent(Config config) {
+        Config context = config.get("context");
+        StaticContentSupport.Builder pBuilder = StaticContentSupport.builder(config.get("location").as(Path.class).get());
+        config.get("welcome")
+                .asString()
+                .ifPresent(pBuilder::welcomeFileName);
+        StaticContentSupport staticContent = pBuilder.build();
+
+        if (context.exists()) {
+            routingBuilder.register(context.asString().get(), staticContent);
+        } else {
+            routingBuilder.register(staticContent);
+        }
+    }
+
+    private void registerClasspathStaticContent(Config config) {
+        Config context = config.get("context");
+
+        StaticContentSupport.Builder cpBuilder = StaticContentSupport.builder(config.get("location").asString().get());
+        cpBuilder.welcomeFileName(config.get("welcome")
+                                          .asString()
+                                          .orElse("index.html"));
+        StaticContentSupport staticContent = cpBuilder.build();
+
+        if (context.exists()) {
+            routingBuilder.register(context.asString().get(), staticContent);
+        } else {
+            routingBuilder.register(staticContent);
+        }
+    }
+
+    private void stopServer(@Observes @Priority(1) @BeforeDestroyed(ApplicationScoped.class) Object event) {
+        if (null == webserver) {
+            // nothing to do
             return;
         }
+        long beforeT = System.nanoTime();
```

Comment:
I've noticed elsewhere that you measure time using `currentTimeMillis()`; I'd pick one and stick with it.  This is probably the right thing to do.

Reply:
Agreed, we can align all time measurements to use `nanoTime`.